File: src/wezterm.ts
import { execFile } from "node:child_process";
import { promisify } from "node:util";

const execFileAsync = promisify(execFile);

export async function wezterm(...args: string[]): Promise<string> {
  const { stdout } = await execFileAsync("wezterm", ["cli", ...args]);
  return stdout;
}

File: src/wezterm.rs
use tokio::process::Command;

use crate::errors::Error;

pub async fn exec(args: &[&str]) -> Result<String, Error> {
    let output = Command::new("wezterm")
        .arg("cli")
        .args(args)
        .output()
        .await?;

    if !output.status.success() {
        let stderr = String::from_utf8_lossy(&output.stderr);
        return Err(Error::Cli(stderr.trim().to_string()));
    }

    Ok(String::from_utf8_lossy(&output.stdout).to_string())
}
